x-pack/plugins/apm/public/components/shared/Links/DiscoverLinks/DiscoverLink.tsx

```diff
@@ -4,27 +4,39 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
+import { EuiLink } from '@elastic/eui';
 import React from 'react';
-import { KibanaRisonLink } from '../KibanaRisonLink';
-import { RisonAPMQueryParams } from '../rison_helpers';
-import { QueryWithIndexPattern } from './QueryWithIndexPattern';
+import chrome from 'ui/chrome';
+import url from 'url';
+import { useAPMIndexPattern } from '../../../../hooks/useAPMIndexPattern';
+import { useLocation } from '../../../../hooks/useLocation';
+import { getRisonString, RisonDecoded } from '../rison_helpers';
 
 interface Props {
-  query: RisonAPMQueryParams;
+  query: RisonDecoded;
   children: React.ReactNode;
 }
 
 export function DiscoverLink({ query, ...rest }: Props) {
-  return (
-    <QueryWithIndexPattern query={query}>
-      {queryWithIndexPattern => (
-        <KibanaRisonLink
-          pathname={'/app/kibana'}
-          hash={'/discover'}
-          query={queryWithIndexPattern}
-          {...rest}
-        />
-      )}
-    </QueryWithIndexPattern>
-  );
+  const apmIndexPattern = useAPMIndexPattern();
+  const location = useLocation();
+
+  if (!apmIndexPattern.id) {
+    return null;
+  }
+
+  const risonSearch = getRisonString(location.search, {
+    ...query,
+    _a: {
+      ...query._a,
+      index: apmIndexPattern.id
+    }
```

Comment:
Don't you need to call `rison.encode` on the `_a` argument?

Reply:
nvm - I asumme that's taken care of by `getRisonString`. As mentioned above I think this could be replaced by:
```js
rison.encode({
  _g: getTimepickerAsRison({
    rangeFrom,
    rangeTo,
    refreshPaused,
    refreshInterval
  }),
  _a: {
    ...query._a,
    index: apmIndexPattern.id
  }
})
```

Reply:
My example above feels a bit verbose but it makes it more clear for the reader what's happening, compared to calling `getRisonString(location.search, query)`

Reply:
I'm tracking what you're saying but took a slightly different approach on fixing it just now. I'll push in a few minutes. Basically, you have to `rison.encode` the `_g` and `_a` values separately and not as a single object, because you need to end up with a string that is `_g={rison-encoded _g}&_a={rison-encoded a}`.

Reply:
Ah, true. Still, I'd just let the link handler to that.